airflow/models/pool.py

```diff
@@ -48,6 +50,45 @@ def get_pool(pool_name, session=None):
     def get_default_pool(session=None):
         return Pool.get_pool(Pool.DEFAULT_POOL_NAME, session=session)
 
+    @staticmethod
+    @provide_session
+    def slots_stats(session=None) -> Dict[str, Dict[str, int]]:
+        from airflow.models.taskinstance import TaskInstance  # Avoid circular import
+
+        pools = {}
+
+        pool_rows = session.query(Pool.pool, Pool.slots).all()
```

Comment:
Do we need to do a second (well, first) query here?

Reply:
Oh I guess we do to get total slots and info about pools that don't have any running tasks. :+1: